### PR TITLE
Fixed various issues with compiling with VS2012

### LINKED ISSUE #696: Bootstrap fails with Visual Studio 2012
The bootstrap makefile fails when using VS 2012. The command `nmake -f Bootstrap.mak MSDEV=vs2012 windows` produces:

```
src/host/os_getversion.c(88) : error C2275: 'HKEY' : illegal use of this type as an expression
        C:\Program Files (x86)\Windows Kits\8.0\include\shared\minwindef.h(239) : see declaration of 'HKEY'
src/host/os_getversion.c(88) : error C2146: syntax error : missing ';' before identifier 'key'
src/host/os_getversion.c(88) : error C2065: 'key' : undeclared identifier
src/host/os_getversion.c(89) : error C2065: 'key' : undeclared identifier
src/host/os_getversion.c(89) : warning C4047: 'function' : 'PHKEY' differs in levels of indirection from 'int *'
src/host/os_getversion.c(89) : warning C4024: 'RegOpenKeyExA' : different types for formal and actual parameter 5
src/host/os_getversion.c(94) : error C2065: 'key' : undeclared identifier
src/host/os_getversion.c(94) : warning C4047: 'function' : 'HKEY' differs in levels of indirection from 'int'
src/host/os_getversion.c(94) : warning C4024: 'RegQueryValueExA' : different types for formal and actual parameter 1
src/host/os_getversion.c(95) : error C2065: 'key' : undeclared identifier
src/host/os_getversion.c(95) : warning C4047: 'function' : 'HKEY' differs in levels of indirection from 'int'
src/host/os_getversion.c(95) : warning C4024: 'RegCloseKey' : different types for formal and actual parameter 1
```

The bootstrap *does* work on VS 2013 and 2015.

### PR DESCRIPTION
**Pull request by samsinsane**
I was able to somehow target v110 with VS2015, but it wasn't able to do the full build, so I may have missed things.

Resolves #696

### CODE REVIEW

File: src/host/os_chdir.c [hunk 1]
```diff
@@ -26,6 +24,8 @@ int do_chdir(lua_State* L, const char* path)
 	z = !chdir(path);
 #endif
 
+	(void)(L);  /* warning: unused parameter */
```
Comment (tvandijck) [inline src/host/os_chdir.c]: if we don't use it... why is it passed in?? I think we should just remove this argument entirely....
bigger change, but probably better long term. I think this is just a remnant for a previous PR.

File: src/host/os_isfile.c [hunk 2]
```diff
@@ -42,5 +41,7 @@ int do_isfile(lua_State* L, const char* filename)
 	}
 #endif
 
+	(void)(L);  /* warning: unused parameter */
```
Comment (tvandijck) [inline src/host/os_isfile.c]: same thing... if not used, don't pass it in..

**Comment by tvandijck** [comment]
Approved, but I think further changes/cleanups are possible.

File: src/host/os_chdir.c (hunk 1 above)
Comment (samsinsane) [inline src/host/os_chdir.c]: For some strange reason, only Windows uses it for an error message. I thought it was very strange, I probably should have scoped it to the non-Windows path, I can make that change if you'd like? :)

Comment (tvandijck) [inline src/host/os_chdir.c]: Ah yes, I see.... yeah, probably best to put it in an #else then.

**Pull request merged**